### ISSUE TITLE
Debug query of source.cubes()

### ISSUE THREAD

**Issue by ptbrowne** [zazuko/rdf-cube-view-query#60]
At the moment it does not seem possible to print the sparql query that is executed by a source.cubes() call.
It could be nice to separate the building of the query from its execution so that we can print the query while developing.

```
const cubesQuery = source.cubes()
console.log(cubesQuery.toString()) // print the query for debug purposes
const cubes = await cubesQuery.execute()
```

**Comment by bergos** [comment]
Since the intermediate object doesn't have any value for other use cases but would add complexity to the API call, I propose adding a new method to the `Source` that returns the generated query.

```javascript
// logs the query for listing the cubes
console.log(source.cubesQuery().toString())

// lists the cubes
const cubes = source.cubes()
```

**Comment by ptbrowne** [comment]
> Since the intermediate object doesn't have any value for other use cases

I think it could have potential other usecases. 

- Getting a query from another place and adding more filters to it
- Serializing the query

I think separating building the query from execution would be good in term of composability. I think having the two concepts as one API call is more complex in this case.

With your example, I am thinking "is it possible for take the output of cubesQuery() and make sure that the exact same query is executed when I call source.cubes() ? Should a source be able to consume the output of cubesQuery() ? Will source.cubes() rely on cubesQuery() in its implementation ?".

**Comment by bergos** [comment]
Some background on `View` vs. `.cubes()`: The `View` class is a wrapper for triples defined in [rdf-cube-view-schema](https://github.com/zazuko/rdf-cube-view-schema). The `.cubes()` method is just a util method and can't be serialized to an RDF representation. The API of this package follows the concept of the rdf-cube-views and provides matching classes. Therefore I would like to stick to my proposal. 

Answers to your questions:

- `.cubes()` will use `.cubesQuery()` to build the query. There will be no redundant code and `.cubesQuery()` will return the same query that is used in `.cubes()`.
- `.cubesQuery()` will be a debug method. The result can't be fed to `.cubes()`. If you have some custom logic to find the terms of cubes, it's quite easy to create a `Cube` instance: Just call `.cube(term)`.
- If serializing the query means to convert it to a string, yes, that's possible with `.toString()`.

**Comment by cristianvasquez** [comment]
This seems to be already implemented: https://github.com/zazuko/rdf-cube-view-query/blob/2321e607e88fa5822476867c2a91ec9cf4143aa5/lib/Source.js#L62. Shall we close this issue?

**Comment by ptbrowne** [comment]
Yes it is possible to get the query text by using cubesQuery but it does not seem very ergonomic to me, and would prefer to have building of the query separate to its execution as stated above, as it would be better composable and less error prone. While debugging, a user must pass the same exact options to cubesQuery and cubes for the debug to be correct.
If execution and building are separated, this is no longer a problem. Since @bergos does not agree, I think we can close.
